projects/plugins/jetpack/_inc/lib/debugger/class-jetpack-cxn-tests.php

```diff
@@ -471,17 +471,56 @@ protected function test__identity_crisis() {
 	}
 
 	/**
-	 * Tests blog and current user's token against wp.com's check-token-health endpoint.
+	 * Tests the health of the Connection tokens.
+	 *
+	 * This will always check the blog token health. It will also check the user token health if
+	 * a user is logged in and connected, or if there's a connected owner.
 	 *
 	 * @since 9.0.0
+	 * @since 9.6.0 Checks only blog token if current user not connected or site does not have a connected owner.
 	 *
 	 * @return array Test results.
 	 */
 	protected function test__connection_token_health() {
-		$name = __FUNCTION__;
+		$m       = new Connection_Manager();
+		$user_id = get_current_user_id() ? get_current_user_id() : $m->get_connection_owner_id();
+
+		if ( $user_id && $m->is_user_connected( $user_id ) ) {
```

Comment:
I don't think this is 100% correct, as it would pass the tests successfully in case of a missing user token.
I know this makes things a bit more complex but `$m->is_user_connected( $user_id )` will return `false` if the user token was deleted.
Maybe we can at least ensure this will test the owner token in case of full connections?

Reply:
p1616768736081700-slack-CBG1CP4EN?thread_ts=1616762306.073100&cid=CBG1CP4EN